lib/pbxProject.js

```diff
@@ -1297,7 +1317,8 @@ pbxProject.prototype.addToOtherLinkerFlags = function (flag) {
     for (config in configurations) {
         buildSettings = configurations[config].buildSettings;
 
-        if (unquote(buildSettings['PRODUCT_NAME']) != this.productName)
+        var productName = opt && opt.productName ? opt.productName : this.productName;
+        if (unquote(buildSettings['PRODUCT_NAME']) != productName)
```

Comment:
Please use strict equal checks, I know these were there before but I personally prefer them

```suggestion
        if (unquote(buildSettings['PRODUCT_NAME']) !== productName)
```